### ISSUE TITLE
Database.init reading metadata each time I want to use ktorm on a connection

### ISSUE THREAD

**Issue by jarekratajski** [kotlin-orm/ktorm#263]
I have a framework where I manage transactions and connections. 
So at the end I have functions which have access to jdbc Connection object.

I want to use ktorm.
Thanks to `Database.connect  { ` I can 
// just have to do stupid wrapping on connection to prevent close

However, I am worried that each time I create database object - the Database object is initialized and in init there is quite a lot of metadata read. This sound like a lot of redundant work if I just call my function multiple times (like on REST service) - each time I am actually using same physical database, same connection pool, just different Connection object - but as for now I cannot reuse Database object.

Would be great if Database class had something like `withConnection`  or `withTransactionManager` where only the current connection was redefined and metadata was reused.

**Comment by vincentlauvlwj** [comment]
Yes, `Database.init` will read metadata each time, but why don't you use the `Database` instance as a long lifecycle object to prevent frequent object creation?

**Comment by jarekratajski** [comment]
`Database` object does not  have a method like `withConnection`. I cannot use connection created by `Database` as it would NOT be associated with a proper database transaction.
